src/Sentry/Hint.cs

```diff
@@ -0,0 +1,143 @@
+using System;
+using System.Collections.Generic;
+using System.Net.Mail;
+using Sentry.Internal.Extensions;
+
+namespace Sentry;
+
+/// <summary>
+/// A hint that can be provided when capturing a <see cref="SentryEvent"/> or adding a <see cref="Breadcrumb"/>.
+/// Hints can be used to filter or modify events or breadcrumbs before they are sent to Sentry.
+/// </summary>
+public class Hint : ICollection, IEnumerable<KeyValuePair<string, object?>>
+{
+    private readonly Dictionary<string, object?> _internalStorage = new();
+    private readonly List<Attachment> _attachments = new();
```

Comment:
I think this class is really blurry, as it encapsulates both a dictionary of items and a list of attachments.  The interfaces it provides are only about the items.  So methods like `Clear`, `Count`, CopyTo`, etc. don't consider the attachments.

I'd rather go with a cleaner data structure:

```csharp
public class Hint
{
    public IDictionary<string, object?> Items { get; } = new Dictionary<string, object?>();
    public IList<Attachment> Attachments { get; } = new List<Attachment>();
}
```

It doesn't need to implement any interfaces of its own.  One will need to be more explicit when using it (ie. `hint.Items["foo"] = bar` - but I think that's ok.

Reply:
This also removes the need for us to have any of our own add/get/remove methods.

Reply:
Yeah I'm OK with that. I pretty much mimicked the Java implementation but agree, it's not clear what things like the "Clear" method do without inspecting their implementation...